File: manual/java-src/com/atta/mcpanel/overlay/tabs/ControlsTab.java
package com.atta.mcpanel.overlay.tabs;

import android.content.Context;
import android.text.InputType;
import android.view.View;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.ScrollView;
import android.widget.TextView;

import com.atta.mcpanel.overlay.PanelHost;
import com.atta.mcpanel.overlay.UiKit;

/**
 * زبانهٔ «ابزارها» — گیم‌مود، زمان، آب‌وهوا و دستور دلخواه (بدراک/جاوا)
 * همهٔ دکمه‌ها یک‌کلیکی: اگر اتصال RCON ذخیره شده باشد، خودکار وصل و اجرا می‌شود.
 */
public final class ControlsTab {

    private ControlsTab() {}

    public static View build(final PanelHost host) {
        final Context ctx = host.getContext();
        final LinearLayout col = UiKit.vcol(ctx, 14);

        col.addView(UiKit.caption(ctx,
                "⚡ یک‌کلیک: اگر اطلاعات سرورت ذخیره باشد، خودکار وصل و دستور اجرا می‌شود.",
                false));

        // ---- گیم‌مود
        col.addView(UiKit.sectionLabel(ctx, "حالت بازی (GameMode)"));
        UiKit.chipRow(ctx, new TextView[]{
                UiKit.chip(ctx, "🟢 خلاق", UiKit.KIND_ACCENT, new Runnable() {
                    @Override public void run() { host.quickCommand("/gamemode creative {p}", true); }
                }),
                UiKit.chip(ctx, "🔵 بقا", new Runnable() {
                    @Override public void run() { host.quickCommand("/gamemode survival {p}", true); }
                }),
                UiKit.chip(ctx, "🟠 ماجراجویی", new Runnable() {
                    @Override public void run() { host.quickCommand("/gamemode adventure {p}", true); }
                }),
        }, host.cols(3), col);

        // ---- زمان
        col.addView(UiKit.sectionLabel(ctx, "زمان روز"));
        UiKit.chipRow(ctx, new TextView[]{
                UiKit.chip(ctx, "🌅 سپیده‌دم", new Runnable() {
                    @Override public void run() { host.quickCommand("/time set 0", true); }
                }),
                UiKit.chip(ctx, "☀️ ظهر", new Runnable() {
                    @Override public void run() { host.quickCommand("/time set 6000", true); }
                }),
                UiKit.chip(ctx, "🌇 غروب", new Runnable() {
                    @Override public void run() { host.quickCommand("/time set 13000", true); }
                }),
                UiKit.chip(ctx, "🌙 نیمه‌شب", new Runnable() {
                    @Override public void run() { host.quickCommand("/time set 18000", true); }
                }),
        }, host.cols(4), col);

        // ---- آب‌وهوا
        col.addView(UiKit.sectionLabel(ctx, "آب‌وهوا"));
        UiKit.chipRow(ctx, new TextView[]{
                UiKit.chip(ctx, "☀️ صاف", new Runnable() {
                    @Override public void run() { host.quickCommand("/weather clear", true); }
                }),
                UiKit.chip(ctx, "🌧️ باران", new Runnable() {
                    @Override public void run() { host.quickCommand("/weather rain", true); }
                }),
                UiKit.chip(ctx, "⛈️ رعدوبرق", new Runnable() {
                    @Override public void run() { host.quickCommand("/weather thunder", true); }
                }),
                UiKit.chip(ctx, "🌦 تغییر هوا", new Runnable() {
                    @Override public void run() { host.quickCommand("/toggledownfall", true); }
                }),
        }, host.cols(4), col);

        // ---- اکشن‌های فوری اپراتور
        col.addView(UiKit.sectionLabel(ctx, "اکشن‌های فوری", false));
        UiKit.chipRow(ctx, new TextView[]{
                UiKit.chip(ctx, "🛡 شب سخت (همه)", UiKit.KIND_DANGER, new Runnable() {
                    @Override public void run() {
                        host.quickCommand("/time set 13000", true);
                        host.quickCommand("/weather thunder", true);
                        host.quickCommand("/effect @a slowness 15 1", true);
                    }
                }),
                UiKit.chip(ctx, "💾 ذخیرهٔ فوری", new Runnable() {
                    @Override public void run() { host.sendConsole("save-all"); }
                }),
                UiKit.chip(ctx, "👥 فهرست بازیکن‌ها", new Runnable() {
                    @Override public void run() { host.sendConsole("list"); }
                }),
        }, host.cols(3), col);

        // ---- دستور دلخواه
        col.addView(UiKit.sectionLabel(ctx, "دستور دلخواه", false));
        final LinearLayout sendRow = UiKit.hrow(ctx);
        final EditText customEt = host.newInput(
                "مثلاً: /effect @a speed 60 2  (بدراک/جاوا)", InputType.TYPE_CLASS_TEXT);
        sendRow.addView(customEt, new LinearLayout.LayoutParams(
                0, UiKit.dp(ctx, 48), 1f));
        TextView send = UiKit.chip(ctx, "اجرا", UiKit.KIND_ACCENT, new Runnable() {
            @Override public void run() {
                host.quickCommand(customEt.getText().toString(), true);
            }
        });
        sendRow.addView(send, new LinearLayout.LayoutParams(
                UiKit.dp(ctx, 88), UiKit.dp(ctx, 48)));
        col.addView(sendRow, UiKit.wrapParams(sendRow, 4, 48));

        ScrollView sv = new ScrollView(ctx);
        sv.setFillViewport(true);
        sv.setOverScrollMode(View.OVER_SCROLL_NEVER);
        sv.addView(col);
        return sv;
    }
}

File: app/src/main/java/com/atta/mcpanel/overlay/tabs/ControlsTab.kt
package com.atta.mcpanel.overlay.tabs

import android.text.InputType
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.ScrollView
import com.atta.mcpanel.overlay.PanelHost
import com.atta.mcpanel.overlay.UiKit

/**
 * زبانهٔ «ابزارها»: ابزارها و لوازم سریع برای اپراتور در حین بازی.
 * همهٔ دستورها با {p} به بازیکن هدف می‌روند؛ بدون اتصال RCON در کلیپ‌بورد کپی
 * می‌شوند تا در چت بازی Paste شوند (مخصوص دنیا/سرور خود اپراتور).
 */
object ControlsTab {

    fun build(host: PanelHost): android.view.View {
        val ctx = host.context
        val col = LinearLayout(ctx).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(UiKit.dp(ctx, 14), UiKit.dp(ctx, 6), UiKit.dp(ctx, 14), UiKit.dp(ctx, 24))
        }

        // ---- بازیکن هدف -------------------------------------------------
        col.addView(UiKit.sectionLabel(ctx, "بازیکن هدف · {p}"))
        val targetEt = host.newInput(
            "نام بازیکن (در سرور خودت) یا @s",
            InputType.TYPE_CLASS_TEXT,
            host.prefs.playerName,
        )
        col.addView(
            targetEt,
            LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT, UiKit.dp(ctx, 46)
            ).apply {
                val m = UiKit.dp(ctx, 4)
                setMargins(m, 0, m, 0)
            }
        )
        col.addView(
            UiKit.caption(
                ctx,
                "دستورها روی این بازیکن اعمال می‌شود؛ اگر وصل به سرور باشی مستقیم اجرا می‌شود، وگرنه در چت بازی Paste می‌کنی."
            )
        )

        fun saveTarget() {
            host.prefs.playerName = targetEt.text.toString()
        }

        fun fire(raw: String, chatOk: Boolean) {
            saveTarget()
            host.quickCommand(host.resolve(raw), chatOk)
        }

        // ---- گیم‌مود -----------------------------------------------------
        col.addView(UiKit.sectionLabel(ctx, "حالت بازی (گیم‌مود)"))
        UiKit.chipRow(
            ctx,
            listOf(
                UiKit.chip(ctx, "خلاق", UiKit.ChipKind.ACCENT) { fire("/gamemode creative {p}", true) },
                UiKit.chip(ctx, "بقا") { fire("/gamemode survival {p}", true) },
                UiKit.chip(ctx, "ماجراجویی") { fire("/gamemode adventure {p}", true) },
                UiKit.chip(ctx, "ناظر") { fire("/gamemode spectator {p}", true) },
            ),
            4, col,
        )

        // ---- هوا ---------------------------------------------------------
        col.addView(UiKit.sectionLabel(ctx, "آب‌وهوا"))
        UiKit.chipRow(
            ctx,
            listOf(
                UiKit.chip(ctx, "☀️ صاف") { fire("/weather clear", true) },
                UiKit.chip(ctx, "🌧️ باران") { fire("/weather rain", true) },
                UiKit.chip(ctx, "⛈️ رعدوبرق") { fire("/weather thunder", true) },
            ),
            3, col,
        )

        // ---- زمان --------------------------------------------------------
        col.addView(UiKit.sectionLabel(ctx, "زمان روز"))
        UiKit.chipRow(
            ctx,
            listOf(
                UiKit.chip(ctx, "🌅 سپیده‌دم") { fire("/time set 0", true) },
                UiKit.chip(ctx, "☀️ ظهر") { fire("/time set 6000", true) },
                UiKit.chip(ctx, "🌇 غروب") { fire("/time set 13000", true) },
                UiKit.chip(ctx, "🌙 نیمه‌شب") { fire("/time set 18000", true) },
            ),
            4, col,
        )

        // ---- آیتم‌ها ------------------------------------------------------
        col.addView(UiKit.sectionLabel(ctx, "لوازم و آیتم‌ها (به بازیکن هدف)"))
        UiKit.chipRow(
            ctx,
            listOf(
                UiKit.chip(ctx, "الماس ×64") { fire("/give {p} diamond 64", true) },
                UiKit.chip(ctx, "ندرایت ×16") { fire("/give {p} netherite_ingot 16", true) },
                UiKit.chip(ctx, "زمرد ×32") { fire("/give {p} emerald_block 32", true) },
                UiKit.chip(ctx, "شمش طلا ×32") { fire("/give {p} gold_ingot 32", true) },
                UiKit.chip(ctx, "سیب طلایی ×16") { fire("/give {p} golden_apple 16", true) },
                UiKit.chip(ctx, "توتم ×8") { fire("/give {p} totem_of_undying 8", true) },
                UiKit.chip(ctx, "ایلیترا") { fire("/give {p} elytra 1", true) },
                UiKit.chip(ctx, "بیکن ×2") { fire("/give {p} beacon 2", true) },
            ),
            4, col,
        )

        // ---- افکت‌ها ------------------------------------------------------
        col.addView(UiKit.sectionLabel(ctx, "افکت‌ها (۶۰ ثانیه)"))
        UiKit.chipRow(
            ctx,
            listOf(
                UiKit.chip(ctx, "سرعت III") { fire("/effect {p} speed 60 2", true) },
                UiKit.chip(ctx, "شب‌بینی") { fire("/effect {p} night_vision 60 0", true) },
                UiKit.chip(ctx, "بازیابی II") { fire("/effect {p} regeneration 60 1", true) },
                UiKit.chip(ctx, "قدرت II") { fire("/effect {p} strength 60 1", true) },
                UiKit.chip(ctx, "مقاومت III") { fire("/effect {p} resistance 60 2", true) },
                UiKit.chip(ctx, "پرش III") { fire("/effect {p} jump_boost 60 2", true) },
                UiKit.chip(ctx, "تنفس زیر آب") { fire("/effect {p} water_breathing 60 0", true) },
                UiKit.chip(ctx, "نامرئی") { fire("/effect {p} invisibility 60 0", true) },
                UiKit.chip(ctx, "پاک‌کردن همه", UiKit.ChipKind.DANGER) {
                    fire("/effect {p} clear", true)
                },
            ),
            3, col,
        )

        // ---- دستور دلخواه ------------------------------------------------
        col.addView(UiKit.sectionLabel(ctx, "دستور دلخواه", accent = false))
        val sendRow = LinearLayout(ctx).apply {
            orientation = LinearLayout.HORIZONTAL
        }
        val customEt: EditText = host.newInput("مثلاً /time set night", InputType.TYPE_CLASS_TEXT)
        sendRow.addView(
            customEt,
            LinearLayout.LayoutParams(0, UiKit.dp(ctx, 46), 1f).apply {
                val m = UiKit.dp(ctx, 4)
                setMargins(m, 0, m, 0)
            }
        )
        sendRow.addView(
            UiKit.chip(ctx, "ارسال", UiKit.ChipKind.ACCENT) {
                fire(customEt.text.toString(), true)
            },
            LinearLayout.LayoutParams(UiKit.dp(ctx, 78), UiKit.dp(ctx, 46)).apply {
                val m = UiKit.dp(ctx, 4)
                setMargins(0, 0, m, 0)
            }
        )
        col.addView(sendRow)

        return ScrollView(ctx).apply {
            isFillViewport = true
            addView(col)
        }
    }
}
